src/erlcloud_guardduty.erl

```diff
@@ -0,0 +1,144 @@
+
+-spec new(AccessKeyID :: string(), SecretAccessKey :: string(), Host :: string()) -> aws_config().
+new(AccessKeyID, SecretAccessKey, Host) ->
+    #aws_config{access_key_id=AccessKeyID,
+                secret_access_key=SecretAccessKey,
+                guardduty_host=Host}.
+
+-spec configure(AccessKeyID :: string(), SecretAccessKey :: string()) -> ok.
+configure(AccessKeyID, SecretAccessKey) ->
+    put(aws_config, new(AccessKeyID, SecretAccessKey)),
+    ok.
+
+-spec configure(AccessKeyID :: string(), SecretAccessKey :: string(), Host :: string()) -> ok.
+configure(AccessKeyID, SecretAccessKey, Host) ->
+    put(aws_config, new(AccessKeyID, SecretAccessKey, Host)),
+    ok.
+
+%%------------------------------------------------------------------------------
+%% API
+%%------------------------------------------------------------------------------
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% GuardDuty API:
+%% [https://docs.aws.amazon.com/guardduty/latest/ug/get-detector.html]
+%%
+%%------------------------------------------------------------------------------
+
+-spec get_detector(DetectorId :: binary()) -> gd_return().
+get_detector(DetectorId) ->
+    get_detector(DetectorId, default_config()).
+
+-spec get_detector(DetectorId :: binary(),
+                   Config     :: aws_config()) -> gd_return().
+get_detector(DetectorId, Config) ->
+    Path = base_path() ++ "detector/" ++ binary_to_list(DetectorId),
+    guardduty_request(Config, get, Path, undefined).
+
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% GuardDuty API:
+%% [https://docs.aws.amazon.com/guardduty/latest/ug/list-detectors.html]
+%%
+%%------------------------------------------------------------------------------
+
+-spec list_detectors() -> gd_return().
+list_detectors() -> list_detectors(default_config()).
+
+-spec list_detectors(Config :: aws_config()) -> gd_return().
+list_detectors(Config) ->
+    list_detectors(undefined, undefined, Config).
+
+-spec list_detectors(Marker   :: binary(),
+                     MaxItems :: integer()) -> gd_return().
+list_detectors(Marker, MaxItems) ->
+    list_detectors(Marker, MaxItems, default_config()).
+
+-spec list_detectors(Marker   :: binary(),
+                     MaxItems :: integer(),
+                     Config   :: aws_config()) -> gd_return().
+list_detectors(Marker, MaxItems, Config) ->
+    Path = base_path() ++ "detector",
+    QParams = filter_undef([{"Marker", Marker},
+                            {"MaxItems", MaxItems}]),
+    guardduty_request(Config, get, Path, undefined, QParams).
+
+
+%%%------------------------------------------------------------------------------
+%%% Internal Functions
+%%%------------------------------------------------------------------------------
+
+filter_undef(List) ->
+    lists:filter(fun({_Name, Value}) -> Value =/= undefined end, List).
```

Comment:
i thought there was a common one for this

Reply:
Nope. The one you are referring to has apparently been declared within `erlcloud_lambda` module. I'll move it to the `erlcloud_util` one and import it from both modules respectively.